containers/Containerfile

```diff
@@ -0,0 +1,45 @@
+# base image off ubuntu image
+ARG UBUNTU_TAG=22.04
+FROM docker.io/ubuntu:${UBUNTU_TAG}
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+# dependencies
+    libboost-all-dev \
+# optional dependencies
+    libtbb-dev \
+    python3-dev \
+    python3-pip \
+    python3-pyparsing \
+    python3-numpy \
+# build dependencies
+    build-essential \
+    cmake \
+# download dependencies
+    git \
+    ca-certificates && \
+    rm -rf /var/lib/apt/lists/*
+
+# build flags
+ARG GTSAM_GIT_TAG=4.2.0
```

Comment:
Could we replace `GTSAM_GIT_TAG` with `GTSAM_VERSION_TAG`? Git tag would refer to the commit hash in a general sense.

Reply:
I see what you mean. That's definitely confusing because you also use PYTHON_TAG etc., so by analogy it's easy to assume you mean version (or in this case a Boolean flag).
Plus, even docker images have tags so it's easy to confuse it with that as well.

I believe every release tag is associated with a version release so the two should be interchangeable.